src/include/processor/execution_context.h

```diff
@@ -6,14 +6,23 @@
 namespace kuzu {
 namespace processor {
 
+class FactorizedTable;
+struct PopulatedCSVError;
+
 struct ExecutionContext {
     uint64_t queryID;
     common::Profiler* profiler;
     main::ClientContext* clientContext;
+    std::shared_ptr<FactorizedTable> warningTable;
+    uint64_t warningLimit;
+    std::mutex mtx;
 
     ExecutionContext(common::Profiler* profiler, main::ClientContext* clientContext,
-        uint64_t queryID)
-        : queryID{queryID}, profiler{profiler}, clientContext{clientContext} {}
+        uint64_t queryID);
+
+    void setWarningMessages(const std::vector<PopulatedCSVError>& messages, uint64_t messageLimit);
+    std::shared_ptr<FactorizedTable>& getWarningTable();
```

Comment:
Why need this `getWarningTable` function? Isn;t all fields public here?

Reply:
good catch, removed